File: plugin-api/sdk/ts/examples/minimal-plugin.ts
import { definePlugin } from "../src/index.js";

export default definePlugin({
  metadata: () => ({
    name: "minimal-plugin",
    version: "0.1.0",
    category: "utility",
    displayName: "Minimal Plugin",
    description: "Minimal make-movie plugin example",
  }),
  initialize: () => undefined,
  shutdown: () => undefined,
});

File: plugin-api/sdk/rust/examples/minimal_plugin.rs
use mm_sdk_rust::{MmPlugin, PluginCategory, PluginMetadata, export_plugin};

#[derive(Default)]
struct MinimalPlugin;

impl MmPlugin for MinimalPlugin {
    fn metadata(&self) -> PluginMetadata {
        PluginMetadata::new("minimal-plugin", "0.1.0", PluginCategory::Utility)
            .display_name("Minimal Plugin")
            .description("Minimal make-movie plugin example")
    }
}

export_plugin!(MinimalPlugin);

fn main() {}
